scripts/DUNE-JP/c101201012.lua

```diff
@@ -0,0 +1,86 @@
+local s,id,o=GetID()
+function s.initial_effect(c)
+	aux.AddCodeList(c,56099748)
+    local e1=Effect.CreateEffect(c)
+	e1:SetDescription(aux.Stringid(id,0))
+    e1:SetCategory(CATEGORY_SPECIAL_SUMMON+CATEGORY_TOHAND+CATEGORY_SEARCH+CATEGORY_GRAVE_ACTION)
+    e1:SetProperty(EFFECT_FLAG_DELAY)
+	e1:SetType(EFFECT_TYPE_FIELD+EFFECT_TYPE_TRIGGER_O)
+    e1:SetCode(EVENT_DESTROYED)
+    e1:SetRange(LOCATION_HAND)
+    e1:SetCountLimit(1,id)
+	e1:SetCondition(s.spcon)
+	e1:SetTarget(s.sptg)
+	e1:SetOperation(s.spop)
+	c:RegisterEffect(e1)
+    local e2=Effect.CreateEffect(c)
+	e2:SetDescription(aux.Stringid(id,1))
+    e2:SetCategory(CATEGORY_ATKCHANGE+CATEGORY_DESTROY)
+    e2:SetProperty(EFFECT_FLAG_DELAY+EFFECT_FLAG_CARD_TARGET)
+    e2:SetType(EFFECT_TYPE_FIELD+EFFECT_TYPE_TRIGGER_O)
+    e2:SetCode(EVENT_DESTROYED)
+    e2:SetRange(LOCATION_MZONE)
+    e2:SetCountLimit(1,id+o)
+    e2:SetCondition(s.descon)
+    e2:SetTarget(s.destg)
+    e2:SetOperation(s.desop)
+    c:RegisterEffect(e2)
+end
+function s.desfilter(c,tp)
+	return c:IsReason(REASON_EFFECT) and c:IsPreviousLocation(LOCATION_ONFIELD)
+end
+function s.confilter(c)
+    return c:IsFaceup() and c:IsCode(56099748)
+end
+function s.spcon(e,tp,eg,ep,ev,re,r,rp)
+	return eg:IsExists(s.desfilter,1,nil,tp) and Duel.IsExistingMatchingCard(s.confilter,tp,LOCATION_ONFIELD,LOCATION_ONFIELD,1,nil)
+end
+function s.sptg(e,tp,eg,ep,ev,re,r,rp,chk)
+	local c=e:GetHandler()
+	if chk==0 then return Duel.GetLocationCount(tp,LOCATION_MZONE)>0 and c:IsCanBeSpecialSummoned(e,0,tp,false,false) end
+	Duel.SetOperationInfo(0,CATEGORY_SPECIAL_SUMMON,c,1,0,0)
+end
+function s.thfilter(c)
+	return c:IsCode(21570001) and c:IsAbleToHand()
+end
+function s.spop(e,tp,eg,ep,ev,re,r,rp)
+	local c=e:GetHandler()
+	if not c:IsRelateToEffect(e) then return end
+	if Duel.SpecialSummon(c,0,tp,tp,false,false,POS_FACEUP)==0 then return end
+	local g=Duel.GetMatchingGroup(aux.NecroValleyFilter(s.thfilter),tp,LOCATION_DECK+LOCATION_GRAVE,0,nil)
+	if #g>0 and Duel.SelectYesNo(tp,aux.Stringid(id,0)) then

```

Comment:
String should be distinct from at https://github.com/Fluorohydride/ygopro-pre-script/pull/1070/files#diff-bc18d332c20500f7f0edfc9cbd114d8bf7d27c3762f2dc7570263dec105d29beR5 
```suggestion
	if #g>0 and Duel.SelectYesNo(tp,aux.Stringid(id,2)) then

```